### PR TITLE
Add sharpening to camera pipe.

### PR DESCRIPTION
**Pull request by dsharletg**
This PR adds a simple unsharp mask to the end of the camera pipe, which makes it a much more interesting pipeline to schedule due to an extra sliding window optimization required.

### CODE REVIEW

File: apps/camera_pipe/Makefile [hunk 1]
```diff
@@ -20,14 +20,14 @@ $(BIN)/viz/camera_pipe.a: $(BIN)/camera_pipe.generator
 	@mkdir -p $(@D)
 	$^ -g camera_pipe -o $(BIN)/viz target=$(HL_TARGET)-trace_loads-trace_stores-trace_realizations
 
-$(BIN)/process: process.cpp $(BIN)/camera_pipe.a $(BIN)/camera_pipe_auto_schedule.a $(BIN)/Demosaic.o $(BIN)/Demosaic_ARM.o
```
Comment (steven-johnson) [inline apps/camera_pipe/Makefile]: The CMakeLists.txt file will need to be updated as well.

File: apps/camera_pipe/camera_pipe_generator.cpp [hunk 2]
```diff
@@ -386,34 +421,41 @@ void CameraPipe::generate() {
...
+            .split(y, yi, yii, 2, TailStrategy::RoundUp)
+            .split(yi, yo, yi, strip_size / 2)
+            .vectorize(x, 2*vec, TailStrategy::RoundUp)
+            .unroll(c)
+            .parallel(yo);
+
         denoised.compute_at(processed, yi).store_at(processed, yo)
             .prefetch(input, y, 2)
-            .fold_storage(y, 8)
+            .fold_storage(y, 16)
```
Comment (steven-johnson) [inline apps/camera_pipe/camera_pipe_generator.cpp]: What are the fold_storage constants based on? Is it worthy pulling them into named constants for clarity?

Comment (dsharletg) [reply apps/camera_pipe/camera_pipe_generator.cpp]: Honestly they're based on trial and error. Stencils combine to produce support requirements that are hard to understand. I'm not sure that named constants help here. When reading the schedule, I feel like the value of the constant matters a lot more than the concept communicated by the name of a constant in these use cases.

**Comment by dsharletg** [comment]
Closing in favor of #2605, rebased on the new master.

**Pull request closed without merging**